File: programmers/shim/Python/level1/소수_찾기_20220623.py
import math
def solution(n):
    answer = 0
    arr=[False]*(n+1)
    for i in range(2, int(math.sqrt(n)+1)):
        if arr[i]==True:
            continue
        for j in range(i+i, n+1,i):
            arr[j]=True
    for i in range(2,n+1):
        if(arr[i]==False):
            answer+=1
    return answer

File: programmers/shim/Java/level1/소수_찾기_20220623.java
package Study_2022.programmers.shim.Java.level1;

import java.util.*;
class Solution {
    public int solution(int n) {
        int answer = 0;
        boolean arr[]=new boolean[n+1];
        Arrays.fill(arr,false);
        for(int i=2; i<=n; i++){
            for(int j=2*i; j<=n; j+=i){
                if(arr[j]==true) continue;
                arr[j]=true;
            }
        }
        for(int i=2; i<=n; i++){
            if(arr[i]==false)
                answer++;
        }
        return answer;
    }
}